CMake/macros.cmake

```diff
@@ -22,6 +22,23 @@ FUNCTION( COPY_DIRECTORY SRC_DIR DST_DIR )
     EXECUTE_PROCESS( COMMAND ${CMAKE_COMMAND} -E copy_directory "${SRC_DIR}" "${DST_DIR}" )
 ENDFUNCTION()
 
+# Create symlinks for a list of files.
+# This uses file(CREATE_LINK ...) when available.
+# Remove function once CMake 3.14 or later is required.
+FUNCTION( SYMLINK_LIST_OF_FILES FILENAMES DST_DIR)
+    IF(${CMAKE_VERSION} VERSION_LESS "3.14")
```

Comment:
3.14 was released in May 2019. So our next release will definitely pass this line with our two-year policy. So feel free to raise the minimum requirement and delete all the work-around for release before 3.14.

Reply:
Actually it was March 2019, already passed 2 year window.

Reply:
I also checked this - we were thinking on the same lines.

However if we can readily not break older versions, we should do so. We could add check + warning for <3.14/2 year old versions though, to prompt users to upgrade.

Reply:
The offload code with Clang also needs 3.14 to make the object target working. I prefer not to spend our time maintaining old cmake workaround which is not covered by our CI or nightly tests anyway. Compilers have longer release cycles and cmake has more frequent release so even more versions to track.

Reply:
Object target is a good point. Combined with this link functionality that is a good forcing function. Lets do it: require minimum cmake 3.14.0.

Reply:
Merge this and tidy up tomorrow?

Reply:
We can do that.